lib/migrate.js

```diff
@@ -338,7 +349,10 @@ CouchDB.prototype.getModelIndexes = function(model, cb) {
       if (item.ddoc !== null) {
         // slice the '_design/'
         var ddocName = item.ddoc.slice(8);
-        var modelIndexName = self.getIndexModelPrefix(mo) + '__' + model;
+        // need to be careful on the subString comparison based on 
+        // naming convertion, avoid one index belongs to two models
+        var modelIndexName = self.getIndexModelPrefix(mo) + '__' + model + '__' +
+          self.getIndexPropertyPrefix(mo);
```

Comment:
Test case for this fix is https://github.com/strongloop/loopback-datasource-juggler/blob/master/test/relations.test.js#L3067

previously the algorithm considered`LBModel__Picture__LBIndex__foo` belongs to both `Picture` and `PictureLink` model, which is wrong.